cognite_toolkit/_cdf_tk/loaders/_data_loaders.py

```diff
@@ -186,7 +187,15 @@ def upload(self, state: "BuildEnvironment", dry_run: bool) -> Iterable[tuple[str
                 # The replacement is used to ensure that we read exactly the same file on Windows and Linux
                 file_content = datafile.read_bytes().replace(b"\r\n", b"\n").decode("utf-8")
                 data = read_csv(io.StringIO(file_content))
-                data.fillna("", inplace=True)
+                # We suppress the pandas FutureWarning about dtype conversion here.
+                # Filling NaNs with an empty string ('') is the desired behavior for uploading to CDF RAW,
+                # even though it converts numeric columns with NaNs to object dtype.
+                # We accept this type change and suppress the warning as it is expected.
+                # This is a temporary measure until pandas is removed as a dependency for this loader.
+                # Note: an additional safeguard is that pandas is capped at < 3.0 in this project.
+                with warnings.catch_warnings():
+                    warnings.filterwarnings("ignore", category=FutureWarning)
```

Comment:
Just cast to object then fill?

Reply:
Then everything becomes string, which is not what we want when uploading to RAW.

Reply:
@doctrino / @ronpal :
----
<img width="1096" height="1676" alt="image" src="https://github.com/user-attachments/assets/41ae27fa-5ca3-4aca-8a6d-bf07ee99003d" />